src/pu-file.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright (c) 2024 PHYTEC Messtechnik GmbH
+ */
+
+#include <gio/gio.h>
+#include "pu-error.h"
+#include "pu-file.h"
+
+gboolean
+pu_file_read_raw(const gchar *filename,
+                 guchar **buffer,
+                 goffset offset,
+                 gssize count,
+                 gsize *bytes_read,
+                 GError **error)
+{
+    g_autoptr(GFile) file = g_file_new_for_path(filename);
+    g_autoptr(GFileInputStream) stream = NULL;
+    g_autoptr(GFileInfo) file_info = NULL;
+    goffset file_size;
+
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    stream = g_file_read(file, NULL, error);
+    if (stream == NULL)
+        return FALSE;
+
+    file_info = g_file_query_info(file, G_FILE_ATTRIBUTE_STANDARD_SIZE,
+                                  G_FILE_QUERY_INFO_NONE, NULL, error);
+    if (file_info == NULL)
+        return FALSE;
+
+    file_size = g_file_info_get_size(file_info);
```

Comment:
Please use `pu_get_file_size()` to get the file size.

Reply:
In this case it is easier to directly use the `g_file_*` functions, as we have direct access to the `GFile` object. Using `pu_get_file_size()` would allocate another `GFile` object just for the size, which is unnecessary.

Reply:
ok, understood.